v2/internal/reconcilers/arm/azure_generic_arm_reconciler_instance.go

```diff
@@ -175,7 +178,9 @@ func (r *azureDeploymentReconcilerInstance) DetermineDeleteAction() (DeleteActio
 	return DeleteActionBeginDelete, r.StartDeleteOfResource, nil
 }
 
-func (r *azureDeploymentReconcilerInstance) DetermineCreateOrUpdateAction() (CreateOrUpdateAction, CreateOrUpdateActionFunc, error) {
+func (r *azureDeploymentReconcilerInstance) DetermineCreateOrUpdateAction(
+	ctx context.Context,
```

Comment:
The original point of this method was to help drive the state-machine we have between create/update or monitor. the reason it didn't take `ctx` was because that decision was supposed to be made based on local state.

Violating that isn't really an issue though, but I am wondering if you considered putting the new code into `BeginCreateOrUpdateResource` instead? I think it would have the same effect there as it does here but would leave this method purely state-machine-management. Possibly we could rename the `BeginCreateorUpdateResource` if you felt its name didn't match the new behavior, although I think it probably still does - it's just that the first part of "CreateOrUpdate" is "Diff" or "Check if we should Create" or whatever.

Reply:
I looked at putting the code into `BeginCreateOrUpdateResource` but figured the intent was to only call that once we'd committed that either create or update was required. 😁 

I'll rejig things to put the code in that method.